.circleci/config.yml

```diff
@@ -196,13 +214,72 @@ commands:
           name: Run puppeteer e2e tests
           working_directory: ~/workbench/e2e
           # parallelism used to run e2e tests
-          command: yarn test:ci $(circleci tests glob "tests/**/*.spec.ts" | circleci tests split --split-by=timings --timings-type=classname --show-counts)
+          command: |
+            circleci tests glob "tests/**/*.spec.ts" | circleci tests split --split-by=timings > /tmp/tests-to-run
+            yarn test:ci $(cat /tmp/tests-to-run)
+          no_output_timeout: 15m
       - store_artifacts:
           path: e2e/logs
           destination: logs
       - store_test_results:
           path: ~/workbench/e2e/logs
 
+
+  start_local_api:
+    description: "Start local API server"
+    steps:
+      - manage_api_cache:
+          restore: true
+      - run:
+          # MySQL sometimes refuses connections by the time we attempt to apply
+          # data migrations. Watch the port for 2m for startup.
+          name: Await MySQL startup
+          command: dockerize -wait tcp://127.0.0.1:3306 -timeout 2m
+      - run:
+          name: Run Local Migrations
+          working_directory: ~/workbench/api
+          command: ./project.rb run-local-migrations
+      - run:
+          name: Launch local API server
+          working_directory: ~/workbench/api
+          # tail -f is important here, it keeps this process running indefinitely
+          command: ./project.rb start-local-api && tail -f build/dev-appserver-out/dev_appserver.out
+          background: true
+      - run:
+          name: Wait for local API server to start
```

Comment:
Note: one optimization you could make, if these things take a while, is to start the servers early, then put the dockerize wait as late as possible. i.e.

```
- start_api
- start_ui
- setup_for_puppeteer
- wait_api
- wait_ui
- puppeteer
```

There's no real need to block immediately

Reply:
That's good strategy. But I think it's best for me that I don't make this change to keep focus only getting local ui + "test" api servers to work.
I will incorporate that in future PR.